// Copyright (C) 2023 The Android Open Source Project
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package wmediumdserver;

import "google/protobuf/empty.proto";

service WmediumdService {
  rpc SetPosition(SetPositionRequest) returns (google.protobuf.Empty) {}
  rpc SetSnr(SetSnrRequest) returns (google.protobuf.Empty) {}
}

message SetPositionRequest {
  string mac_address = 1;
  double x_pos = 2;
  double y_pos = 3;
}

message SetSnrRequest {
  string mac_address_1 = 1;
  string mac_address_2 = 2;
  int32 snr = 3;
}
